hs_core/views/resource_folder_hierarchy.py

```diff
@@ -452,3 +457,199 @@ def data_store_file_or_folder_move_or_rename(request, res_id=None):
+    res_id = str(res_id).strip()
+    try:
+        resource, _, user = authorize(request, res_id,
+                                      needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+    except NotFound:
+        return HttpResponse('Bad request - resource not found', status=status.HTTP_400_BAD_REQUEST)
+    except PermissionDenied:
+        return HttpResponse('Permission denied', status=status.HTTP_401_UNAUTHORIZED)
+
+    src_path = resolve_request(request).get('source_path', None)
+    tgt_path = resolve_request(request).get('target_path', None)
+    if src_path is None or tgt_path is None:
+        return HttpResponse('Bad request - src_path or tgt_path is not included',
+                            status=status.HTTP_400_BAD_REQUEST)
+    src_path = str(src_path).strip()
+    tgt_path = str(tgt_path).strip()
+    if not src_path or not tgt_path:
+        return HttpResponse('Bad request - src_path or tgt_path cannot be empty',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if not src_path.startswith('data/contents/'):
+        return HttpResponse('Bad request - src_path must start with data/contents/',
+                            status=status.HTTP_400_BAD_REQUEST)
+    if src_path.find('/../') >= 0 or src_path.endswith('/..'):
+        return HttpResponse('Bad request - src_path cannot contain /../',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if not tgt_path.startswith('data/contents/'):
+        return HttpResponse('Bad request - tgt_path must start with data/contents/',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if tgt_path.find('/../') >= 0 or tgt_path.endswith('/..'):
+        return HttpResponse('Bad request - tgt_path cannot contain /../',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    # check that the target is a folder in iRODs or doesn't exist
+    # (and will be created as a folder)
+    istorage = resource.get_irods_storage()
+    tgt_folder, tgt_base = os.path.split(tgt_path)
+    tgt_folder = tgt_folder[len('data/contents/'):]
+    abspath = get_path(resource, tgt_base, folder=tgt_folder)
+    if istorage.exists(abspath) and not is_folder(istorage, abspath):
+        return HttpResponse('Bad request - tgt_path is not a folder',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    try:
+        move_to_folder(user, res_id, src_path, tgt_path)
+    except SessionException as ex:
+        return HttpResponse(ex.stderr, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+    except DRF_ValidationError as ex:
+        return HttpResponse(ex.detail, status=status.HTTP_400_BAD_REQUEST)
+
+    return_object = {'target_rel_path': tgt_path}
+
+    return HttpResponse(
+        json.dumps(return_object),
+        content_type='application/json'
+    )
+
+
+@api_view(['POST'])
+def data_store_move_to_folder_public(request, pk):
+    return data_store_move_to_folder(request, res_id=pk)

```

Comment:
This is not necessarily a pattern we need to follow (making a function named function_name, and then a function_name_public). I was just doing that because I was harnessing existing non Django Rest Framework code with proper DRF functions.